gravitee-apim-console-webui/src/management/home/components/gio-api-events-table/gio-api-events-table.component.ts

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright (C) 2015 The Gravitee team (http://gravitee.io)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { Component, Inject, Input, OnChanges, OnDestroy, SimpleChanges } from '@angular/core';
+import { isEqual, toNumber } from 'lodash';
+import { StateService } from '@uirouter/angularjs';
+import { BehaviorSubject, Observable, of, Subject } from 'rxjs';
+import { distinctUntilChanged, map, startWith, switchMap, tap } from 'rxjs/operators';
+
+import { UIRouterState } from '../../../../ajs-upgraded-providers';
+import { GioTableWrapperFilters } from '../../../../shared/components/gio-table-wrapper/gio-table-wrapper.component';
+import { EventService } from '../../../../services-ngx/event.service';
+import { TimeRangeParams } from '../gio-quick-time-range/gio-quick-time-range.component';
+
+type TableDataSource = {
+  apiId: string;
+  apiName: string;
+  apiVersion: string;
+  deploymentNumber: number;
+  updatedAt: Date;
+  deploymentLabel: string;
+  type: string;
+};
+@Component({
+  selector: 'gio-api-events-table',
+  template: require('./gio-api-events-table.component.html'),
+  styles: [require('./gio-api-events-table.component.scss')],
+})
+export class GioApiEventsTableComponent implements OnChanges, OnDestroy {
+  private filters$ = new BehaviorSubject<(GioTableWrapperFilters & TimeRangeParams) | null>(null);
+  private unsubscribe$: Subject<boolean> = new Subject<boolean>();
+
+  @Input()
+  timeRangeParams: TimeRangeParams;
+
+  isLoading = true;
+  totalLength = 0;
+  currentFilters: GioTableWrapperFilters = {
+    searchTerm: '',
+    pagination: { index: 1, size: 5 },
+  };
+  displayedColumns = ['name', 'deployment', 'date', 'type'];
+  tableDataSource$: Observable<TableDataSource[]> = this.filters$.pipe(
+    distinctUntilChanged(isEqual),
+    tap(() => (this.isLoading = true)),
+    switchMap((filters) => {
+      if (filters === null) {
+        return of([]);
+      }
+      const currentFilters = filters;
+      return this.eventService
+        .search(
+          'START_API,STOP_API,PUBLISH_API,UNPUBLISH_API',
+          '',
+          filters.searchTerm,
+          filters.from,
+          filters.to,
+          filters.pagination.index,
+          filters.pagination.size,
+        )
+        .pipe(
```

Comment:
Is it possible to flatten this ?

Reply:
or maybe I'm not seeing the "simplification" 🙈